### ISSUE TITLE
Support Korea region

### ISSUE THREAD

**Issue by Royental** [Hyundai-Kia-Connect/kia_uvo#701]
Could you please make it can support Korean region?
Ironically, your masterpiece cannot be used in Korea.
Please let me know if there is any information I can provide for Korean region support.

Thank you!

**Comment by Royental** [comment]
> Are you able to sniff the traffic for us?

Thank you for your interest.
Any references to learn how to sniff traffic? I'll try.

**Comment by cdnninja** [comment]
Charles proxy seems to work well if a mobile app.  If a website debug mode works too.

**Comment by cdnninja** [comment]
Key is following the flow for urls and headers to understand how it works.  

Then to try align that to one of our existing regions to see if it is the same.  If so we can reuse work.

**Comment by Royental** [comment]
> Key is following the flow for urls and headers to understand how it works.
> 
> Then to try align that to one of our existing regions to see if it is the same. If so we can reuse work.

Thank you cdnninja!

I'll let you know once I get the packets.

**Comment by Royental** [comment]
[bluelink.chls.zip](https://github.com/Hyundai-Kia-Connect/kia_uvo/files/12325300/bluelink.chls.zip)

I got the packets via Charles Proxy.
I hope this data is helpful for support Korean region.

Please let me know if you need any additional information.

Thank you!

**Comment by kjd2338** [comment]
[Genesis_KR_Capture.zip](https://github.com/Hyundai-Kia-Connect/kia_uvo/files/12326581/Genesis_KR_Capture.zip)
I am adding Genesis KR Packets via Charles Proxy.
I hope this data helps too.

Thank you~! ;)

**Comment by cdnninja** [comment]
Looks like SSL proxying wasn't enabled and configured so I can't see the contents of the stream.  As a note once this is done it will contain items like your password.   

This applies to both the Genesis and bluelink capture.

**Comment by Royental** [comment]
The desired outcome wasn't achieved, but thank you for your efforts. 
May good luck always be with you!

**Comment by cdnninja** [comment]
> The desired outcome wasn't achieved, but thank you for your efforts. 
> 
> May good luck always be with you!

Just to clarify this is something you need to enable and redo for me to watch the traffic.

**Comment by Royental** [comment]
> 
Oh, It's not ended.
I'll try it again. :)

**Comment by Royental** [comment]
[Sonata the edge KR.chls.zip](https://github.com/Hyundai-Kia-Connect/kia_uvo/files/12328516/Sonata.the.edge.KR.chls.zip)

I did it again. Please check with this.

Thank you!

**Comment by cdnninja** [comment]
Same thing: 

![image](https://github.com/Hyundai-Kia-Connect/kia_uvo/assets/6373468/82f3ee55-1996-42d2-ab38-c2ea354bb8dc)

It should show header tabs etc on each item

**Comment by Royental** [comment]
[Allow SSL.chls.zip](https://github.com/Hyundai-Kia-Connect/kia_uvo/files/12328552/Allow.SSL.chls.zip)

Thank you cdnninja!
Finally I understood what does that mean.

Could you please check this file?
I've checked the note 

[Notes | SSL Proxying enabled for this host]

**Comment by Royental** [comment]
> Same thing:
> 
> ![image](https://user-images.githubusercontent.com/6373468/260270234-82f3ee55-1996-42d2-ab38-c2ea354bb8dc.png)
> 
> It should show header tabs etc on each item.
> 
> SSL proxying in charles need to be turned on and the SSL cert installed into the android device.

[Sonata the edge.chls.zip](https://github.com/Hyundai-Kia-Connect/kia_uvo/files/12328660/Sonata.the.edge.chls.zip)

I found my problem on Charles proxy and fixed it.
Could you please check this file?

Thank you!

**Comment by cdnninja** [comment]
It does look pretty similar to EU.  This capture doesn't seem to contain the login flow though.  Maybe email me a copy with that? macman64@gmail.com

**Comment by Royental** [comment]
> It does look pretty similar to EU.



> It does look pretty similar to EU. This capture doesn't seem to contain the login flow though. Maybe email me a copy with that? [macman64@gmail.com](mailto:macman64@gmail.com)

I've sent an email include packets.

Thank you!

**Comment by cdnninja** [comment]
So while the endpoints are roughly the same it looks like they don't use something called stamps like EU.  This may be a good thing, but will take some time to adjust to confirm.

**Comment by cdnninja** [comment]
I have started a draft of the code here: https://github.com/Hyundai-Kia-Connect/hyundai_kia_connect_api/pull/388  

Currently it doesn't work,  it gets a device ID no issue but after that fails with no authorization on the next part. I suspect headers and cookies need some work.   I also noticed it has a ccsp-uuid that other regions don't have, it seems to change but not sure how to predict it.

**Comment by corsair-kr** [comment]
hello.

Is the control failing after login?
Or are you having problems logging into your account?

**Comment by cdnninja** [comment]
No haven't made it to login just yet.  It gets device id fine but after that fails with a 401 error code before it gets to login command.   I haven't spent more time on it, feel free to give it a try based on the PR code I have linked.

**Comment by nara1111** [comment]
Hello. It seems that in regions outside of Korea, this API was created through reverse engineering of iPhone and Android apps. However, in Korea, Hyundai Motor Group has put a lot of effort into telematics services and is officially providing API specifications. Of course, since it is not based on an app, it only provides a very limited level of information, such as driving distance status. But is there a way to use this?

**Comment by cdnninja** [comment]
Yes. With docs wouldn't be hard to do.   Are you willing to do the dev if I guide you?

**Comment by nara1111** [comment]
@cdnninja I'm currently trying this with claude.ai, and if it works, it would be a great help. In fact, I was working on this as a personal project to integrate it with HA, but the HA integration also ends up being based on this API, so I need to modify this API. By the way, I don't have much development knowledge, so I'm taking this slowly.

**Comment by cdnninja** [comment]
It would be a largely a single file for this integration to support it. With template to follow.
